### DISCUSSION TITLE
Set popup z coordinate

### DISCUSSION THREAD

**Discussion by dodgytricycle** [uber/react-map-gl#2026] [Q&A] [answered]
I have a PointCloudLayer rendering 3D points and would like to open a Popup when I click on one of those points. However the Popup component only takes latitude and longitude, so if I move or tilt the map the popup anchor appears to move off the selected point. 

Potentially I would need to calculate an offset each time the view state changes. Before I go down that path, does anyone have any suggestions for an easier/simpler/more elegant approach? Cheers

**Comment by Pessimistress** [comment, marked as answer]
If you are using v5/v6, there is an `altitude` prop: https://github.com/visgl/react-map-gl/blob/6.1-release/docs/api-reference/popup.md

**Reply by dodgytricycle** [reply]
Much appreciated :)
